src/pe/relocation.rs

```diff
@@ -131,3 +308,370 @@ impl<'a> Iterator for Relocations<'a> {
+                        "Cannot map base reloc rva {:#x} into offset",
+                        dd.virtual_address
+                    ))
+                })?;
+        let bytes = bytes[offset..]
+            .pread::<&[u8]>(dd.size as usize)
+            .map_err(|_| {
+                error::Error::Malformed(format!(
+                    "base reloc offset {:#x} and size {:#x} exceeds the bounds of the bytes size {:#x}",
+                    offset,
+                    dd.size,
+                    bytes.len()
+                ))
+            })?;
+
+        Ok(Self { bytes })
+    }
+
+    /// Returns iterator for [`RelocationBlock`]
+    pub fn blocks(&self) -> RelocationBlockIterator<'a> {
+        RelocationBlockIterator {
+            bytes: &self.bytes,
+            offset: 0,
+        }
+    }
+}
+
+/// Represents a PE (Portable Executable) base relocation block.
+///
+/// In the PE format, base relocations are organized into blocks, where each block corresponds
+/// to a contiguous range of memory that needs to be relocated. The relocations within a block
+/// specify offsets relative to the block's starting address.
+///
+/// Unlike COFF (Common Object File Format), PE base relocations are separated by blocks.
+/// Each block begins with a [`RelocationBlock`] structure that provides information about the
+/// relocation block, including its starting RVA (Relative Virtual Address) and size in bytes.
+///
+/// Each block ends with an "absolute" relocation entry, which acts as a marker to indicate
+/// the end of the block. This entry has no actual relocation effect.
+#[derive(Debug, Copy, Clone, PartialEq, Default)]
+pub struct RelocationBlock<'a> {
+    /// The starting RVA for the relocation block.
+    pub rva: u32,
+    /// The total size of the relocation block, in bytes.
+    pub size: u32,
+    /// The bytes of [`RelocationWord`] data within this block.
+    pub bytes: &'a [u8],
+}
+
+impl RelocationBlock<'_> {
+    /// Returns iterator for [`RelocationWord`]
+    pub fn words(&self) -> RelocationWordIterator {
+        RelocationWordIterator {
+            bytes: &self.bytes,
+            offset: 0,
+        }
+    }
+}
+
+/// Raw size of [`RelocationBlock`]
+pub const RELOCATION_BLOCK_SIZE: usize = 8;
+
+impl<'a> scroll::ctx::TryFromCtx<'a, scroll::Endian> for RelocationBlock<'a> {
+    type Error = crate::error::Error;
+
+    fn try_from_ctx(bytes: &'a [u8], ctx: scroll::Endian) -> Result<(Self, usize), Self::Error> {
+        let mut offset = 0;
+        let rva = bytes.gread_with::<u32>(&mut offset, ctx)?;
+        let size = bytes.gread_with::<u32>(&mut offset, ctx)?;
+        if (size as usize) < RELOCATION_BLOCK_SIZE {
+            return Err(error::Error::Malformed(format!(
+                "base reloc size {size} is less than a block {RELOCATION_BLOCK_SIZE}"
+            )));
+        }
+        let word_size = size as usize - RELOCATION_BLOCK_SIZE;
+        if offset + word_size > bytes.len() {
```

Comment:
Technically you could do that same pread trick here too but not important

Reply:
Yes I was about to do that here as well; Turns out this expression is more clear for use in here. Specifically `word_size` is a bit confusing stuff to crush into `pread<&[u8]>(word_size)`, as it requrie subtraction of previous `size_of(RelocationBlock::rva)` + `size_of(RelocationBlock::size)`. I'd think this is good for maintainability.

Reply:
Nvm, just got the better expression without confusing around `word_size`. Thanks for suggestion!